### PR TITLE
Component of planet initialization on the map

### PR DESCRIPTION
**Pull request by TheShuEd**
## About the PR
Currently, creating planetary maps (europa, and terminal) requires turning the map into a planet while still mapping. As far as I know, this has several negative consequences:
1) Problems with saving the map. After creating a planet, the file becomes much less stable, and editing planetary maps more often causes errors
2) Impossibility of random generation. The planet grain is selected on mapping, and is repeated from time to time when loading the map. 
3) Inability to change biomes quickly. 
(I could generally be wrong on all three counts altogether, haha)

But, I propose a simple and elegant solution. A component that connect on the map entity, and starts the planet creation process during MapInitEvent. This ensures that the planet doesn't interfere with mapping, promises easy biome switching, and procedural environment generation every round.

:cl:
- tweak: Now winter around Europa is different every round

### CODE REVIEW

**Comment by TheShuEd** [comment]
huh, wat

**Comment by deltanedas** [comment]
dev probably shouldnt be a planet to not interfere with testing anything that needs to be in space

**Comment by TheShuEd** [comment]
Yes, I agree, and that was going to be removed anyway, but now we need to figure out what the error problem is

**Comment by TheShuEd** [comment]
but I noticed that in this implementation the evacuation shuttle flies to the right place, not to a random point on the station, gib half the crew.

File: Content.Server/Parallax/SpawnMapBiomeSystem.cs [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+    {
+        base.Initialize();
+        SubscribeLocalEvent<SpawnMapBiomeComponent, MapInitEvent>(OnMapInit);
+    }
+
+    private void OnMapInit(Entity<SpawnMapBiomeComponent> map, ref MapInitEvent args)
+    {
+        if (!_mapManager.IsMap(map)) return;
+
+        _biome.EnsurePlanet(map, _proto.Index(map.Comp.Biome));
```
[the remark this answers is not in the document]
Comment (TheShuEd) [reply Content.Server/Parallax/SpawnMapBiomeSystem.cs:23]: image

[the remark this answers is not in the document]
Comment (TheShuEd) [reply Content.Server/Parallax/SpawnMapBiomeSystem.cs:23]: it doesn't really seem like it, does it?

**Comment by TheShuEd** [comment]
there's something I can't figure out. Just adding a BiomeComponent to the map breaks the logic?

File: Content.Server/Parallax/SpawnMapBiomeSystem.cs (hunk 1 above)
[the remark this answers is not in the document]
Comment (deltanedas) [reply Content.Server/Parallax/SpawnMapBiomeSystem.cs:23]: if you add this component in vv to a map entity does it work as normally

and to be clear it works on map entity not station entity

[the remark this answers is not in the document]
Comment (TheShuEd) [reply Content.Server/Parallax/SpawnMapBiomeSystem.cs:23]: And i add it to map entity

[the remark this answers is not in the document]
Comment (TheShuEd) [reply Content.Server/Parallax/SpawnMapBiomeSystem.cs:23]: Its works in game in any scenarios, but broke integration test

> if you add this component in vv to a map entity does it work as normally

**Comment by TheShuEd** [comment]
Okay, it's working fine now. It is now a component attached to the station prototype. I removed the planet from Dev, and also cleaned up everything related to the planet in the map file from Europe. Instead, I added a new planet initialization component to Europe. Now the winter around Europa is different every time.

image
and on the mapping it is now back in space, which makes it easier to edit.

File: Content.Server/Station/Components/StationBiomeComponent.cs [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+using Content.Shared.Parallax.Biomes;
+using Robust.Shared.Prototypes;
+
+namespace Content.Server.Station.Components;
+
+/// <summary>
+/// allows you to initialize a planet on a specific map at initialization time.
+/// </summary>
+
+[RegisterComponent, Access(typeof(StationBiomeSystem))]
```
Comment (metalgearsloth) [inline Content.Server/Station/Components/StationBiomeComponent.cs:11]:
```suggestion

/// <summary>
/// Runs EnsurePlanet against the largest grid on Mapinit.
/// </summary>
[RegisterComponent, Access(typeof(StationBiomeSystem))]
```

File: Content.Server/Station/Components/StationBiomeComponent.cs [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+
+/// <summary>
+/// Runs EnsurePlanet against the largest grid on Mapinit.
+/// </summary>
+[RegisterComponent, Access(typeof(StationBiomeSystem))]
+public sealed partial class StationBiomeComponent : Component
+{
+    [DataField]
+    public ProtoId<BiomeTemplatePrototype> Biome = "Grasslands";
+
```
Comment (metalgearsloth) [inline Content.Server/Station/Components/StationBiomeComponent.cs:15]:
```suggestion
{
    [DataField(required: true)]
    public ProtoId<BiomeTemplatePrototype> Biome = "Grasslands";

```

**Pull request merged**